openedx/features/survey_report/api.py

```diff
@@ -0,0 +1,35 @@
+"""
+Contains the logic to manage survey report model.
+"""
+
+from django.conf import settings
+
+from openedx.features.survey_report.models import SurveyReport
+from openedx.features.survey_report.queries import (
+    get_course_enrollments,
+    get_recently_active_users,
+    get_generated_certificates,
+    get_registered_learners,
+    get_unique_courses_offered
+)
+
+
+def generate_report() -> None:
+    """ Generate a report with relevant data."""
+    courses_offered = get_unique_courses_offered()
+    learners = get_recently_active_users(weeks=3)
```

Comment:
Should we make this number of weeks a constant?
Even perhaps a constant that is defined as a django setting for those that want to change it.

I raise the issue with doubt since allowing reporters to change the settings for the report generation makes the results very difficult to compare.

Reply:
Maybe, We can continue with the weeks like an arg because, is more readable and easy to change by developers.

Because, Like just said, change the range by settings makes the results too difficult to compare.

Reply:
I'm not a big fan of hard-coded constants without explanation, we should at least make it a constant with a descriptive name.

Reply:
I agree with @mariajgrimaldi here. If we are leaving the value as a constant, then we should at least make it into a constant at the top of the module where we can also leave some explanation about the value.

The value itself is a threshold just like any other. It was the product of the analisis that was used for the first survey.

However as far as I can tell the value was 4. https://gist.github.com/feanil/0b49d5219ae08baf1216556d455dfdbb#file-instruction-md. A value of 4 would also be easier to understand intuitively, it is the users that where active in the last Month.